### ISSUE TITLE
Valid PolygonWithHoles triggers a precondition fail during orientation check

### ISSUE THREAD

**Issue by Fro-Z** [CGAL/cgal#3697]
I found a valid PolygonWithHoles that triggers a precondition fail during orientation test.
```c++
#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
#include <CGAL/Boolean_set_operations_2.h>
#include <CGAL/Polygon_set_2.h>
#include <sstream>
#include <assert.h>

using K = CGAL::Exact_predicates_exact_constructions_kernel;
using PolygonSet = CGAL::Polygon_set_2<K>;
using PolygonWithHoles = CGAL::Polygon_with_holes_2<K>;

int main()
{
    std::stringstream sstream("6  2 1  3 0  5 4  2 1  1 3  0 2  0");
    PolygonWithHoles poly;
    sstream >> poly;

    assert(CGAL::is_valid_polygon_with_holes(poly, PolygonSet::Traits_2())); //This fails
}
```
This is the polygon:
![img](https://user-images.githubusercontent.com/8879226/53129901-1748dc80-3569-11e9-8833-b70636d1760f.png)

The precondition triggered is one in Arr_segment_traits. (Gets called from Gps_traits_adaptor)

This issue is unrelated to #3667 , and is not fixed by #3688 as of now.

## Environment
* Operating system: Windows 10
* Compiler: MSVC 2017 15.9.4
* Release or debug mode: Tested in debug mode, likely happens in both
* Specific flags used (if any):
  Preprocessor defines:
  CGAL_USE_CORE=1
  CGAL_HEADER_ONLY=1
* CGAL version: 4.13, from the installer
* Boost version: 1.68
* Other libraries versions if used (Eigen, TBB, etc.):

**Comment by efifogel** [comment]
Yep. My previous fix was incorrect.
I've fixed it right this time.

**Comment by lrineau** [comment]
Fixed by #3688.
